drivers/usb_c/tcpc/rt1715.c

```diff
@@ -0,0 +1,712 @@
+		/* Enable external VCONN power supply */
+		gpio_pin_set_dt(&cfg->vconn_ctrl_gpio, true);
+	}
+
+	data->cc_changed = true;
+	ret = tcpci_tcpm_set_vconn(&cfg->bus, enable);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (enable == false && cfg->vconn_ctrl_gpio.port != NULL) {
+		/* Disable external VCONN power supply */
+		gpio_pin_set_dt(&cfg->vconn_ctrl_gpio, false);
+	}
+
+	if (data->vconn_cb != NULL) {
+		ret = data->vconn_cb(dev, data->cc_polarity, enable);
+	}
+
+	return ret;
+}
+
+static int rt1715_tcpc_set_roles(const struct device *dev, enum tc_power_role power_role,
+				 enum tc_data_role data_role)
+{
+	const struct rt1715_cfg *cfg = dev->config;
+
+	return tcpci_tcpm_set_roles(&cfg->bus, PD_REV30, power_role, data_role);
+}
+
+static int rt1715_tcpc_get_rx_pending_msg(const struct device *dev, struct pd_msg *msg)
+{
+	struct rt1715_data *data = dev->data;
+
+	/* Rx message pending? */
+	if (!data->msg_pending) {
+		return -ENODATA;
+	}
+
+	/* Query status only? */
+	if (msg == NULL) {
+		return 0;
+	}
+
+	/* Dequeue Rx FIFO */
+	*msg = data->rx_msg;
+	data->msg_pending = false;
+
+	/* Indicate Rx message returned */
+	return 1;
+}
+
+static int rt1715_tcpc_rx_fifo_enqueue(const struct device *dev)
+{
+	const struct rt1715_cfg *const cfg = dev->config;
+	struct rt1715_data *data = dev->data;
+	uint8_t rxbcnt;
+	uint8_t rxftype;
+	uint16_t rxhead;
+	uint8_t rx_data_size;
+	struct pd_msg *msg = &data->rx_msg;
+	int ret = 0;
+
+	ret |= tcpci_read_reg8(&cfg->bus, TCPC_REG_RX_BUFFER, &rxbcnt);
+	ret |= tcpci_read_reg8(&cfg->bus, TCPC_REG_RX_BUFFER + 1, &rxftype);
+	ret |= tcpci_read_reg16(&cfg->bus, TCPC_REG_RX_BUFFER + 2, &rxhead);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* rxbcnt = 1 (frame type) + 2 (Message Header) + Rx data byte count */
+	if (rxbcnt < 3) {
+		LOG_ERR("Invalid RXBCNT: %d", rxbcnt);
+		return -EIO;
+	}
+	rx_data_size = rxbcnt - 3;
+
+	/* Not support Unchunked Extended Message exceeding PD_CONVERT_PD_HEADER_COUNT_TO_BYTES */
+	if (rx_data_size > (PD_MAX_EXTENDED_MSG_LEGACY_LEN + 2)) {
+		LOG_ERR("Not support Unchunked Extended Message exceeding "
+			"PD_CONVERT_PD_HEADER_COUNT_TO_BYTES: %d",
+			rx_data_size);
+		return -EIO;
+	}
+
+	/* Rx frame type */
+	msg->type = rxftype;
+
+	/* Rx header */
+	msg->header.raw_value = (uint16_t)rxhead;
+
+	/* Rx data size */
+	msg->len = rx_data_size;
+
+	/* Rx data */
+	if (rx_data_size > 0) {
+		ret = i2c_burst_read_dt(&cfg->bus, TCPC_REG_RX_BUFFER + 4, msg->data, rx_data_size);
+		if (ret) {
+			LOG_ERR("Failed to read Rx data: %d", ret);
+		}
+	}
+
+	return ret;
+}
+
+static int rt1715_tcpc_set_rx_enable(const struct device *dev, bool enable)
+{
+	const struct rt1715_cfg *cfg = dev->config;
+	struct rt1715_data *data = dev->data;
+
+	if (enable == false) {
+		return tcpci_tcpm_set_rx_type(&cfg->bus, 0);
+	}
+
+	if (data->rx_sop_prime_enable) {
+		return tcpci_tcpm_set_rx_type(&cfg->bus,
+					      TCPC_REG_RX_DETECT_SOP_SOPP_SOPPP_HRST_MASK);
```

Comment:
Why are you enabling both SOP' and SOP'' messages here?

Reply:
I noticed that both Numaker and STM32 enable them as well. However, I couldn't find the datasheet for the PS8xxx series, so I suspect that it may not support SOP'' messages.

Reply:
The PS8xxx does support SOP' and SOP''.  However, the API documentation for `tcpc_sop_prime_enable()` indicates this command is used to enable SOP' messages, with no mention of SOP''.

The [ChromeOS code](https://source.corp.google.com/h/chrome-internal/chromeos/superproject/+/main:src/platform/ec/driver/tcpm/tcpci.c;l=637;drc=c4c64c80b0a83f646515629312cc67ffa59b4daa) that USB-C support is based on does enable SOP' and SOP'' for this command.

Do you mind updating the documentation for `tcpc_sop_prime_enable()` to note that this enables both SOP' and SOP''?

Reply:
Done. Please review.